runtime/src/main/java/org/corfudb/runtime/collections/TxnStreamingManager.java

```diff
@@ -148,7 +154,13 @@ private synchronized void scheduleTxnPolling() {
         if (!lockedStreamingSubscriptionContexts.isEmpty()) {
             log.trace("Locked {} StreamingSubscriptionContexts for processing",
                     lockedStreamingSubscriptionContexts.size());
-            pollerExecutor.submit(new TransactionPoller(runtime, lockedStreamingSubscriptionContexts));
+
+            // To prevent leaking transaction stream caches, explicitly remove the prior entry
+            if (transactionPoller != null) {
+                runtime.getStreamsView().getOpenedStreams().remove(transactionPoller.getTxnStream());
```

Comment:
Not a big fan of directly removing things from the underlying StreamsView. Am open to other working alternatives or suggestions.